shell/edit/auth/oidc.vue

```diff
@@ -59,42 +68,67 @@ export default {
       };
     },
 
+    validationPassed() {
+      if ( this.editConfig ) {
+        if ( !this.customEndpoint.value ) {
+          return !!(this.model.clientId && this.model.clientSecret && this.oidcUrls.url && this.oidcUrls.realm && this.oidcScope.includes('openid'));
+        } else {
+          return !!(this.model.clientId && this.model.clientSecret && this.model.rancherUrl && this.model.issuer && this.model.authEndpoint && this.oidcScope.includes('openid'));
+        }
+      }
+
+      return true;
+    }
   },
+
   watch: {
-    'keycloakUrls.url'() {
-      this.updateIssuerEndpoint();
+    'oidcUrls.url'() {
+      this.updateEndpoints();
     },
-    'keycloakUrls.realm'() {
-      this.updateIssuerEndpoint();
+
+    'oidcUrls.realm'() {
+      this.updateEndpoints();
     },
+
     'model.enabled'(neu) {
       // Cover case where oidc gets disabled and we return to the edit screen with a reset model
       if (!neu) {
-        this.keycloakUrls = {
-          url:   null,
-          realm: null
+        this.oidcUrls = {
+          url:              null,
+          realm:            null,
+          jwksUrl:          null,
+          tokenEndpoint:    null,
+          userInfoEndpoint: null,
         };
         this.customEndpoint.value = false;
+        this.oidcScope = this.model?.scope?.split(' ');
+      } else {
+        this.oidcScope = this.model?.scope?.split(' ');
       }
     },
+
     editConfig(neu, old) {
-      // Cover use case where user edits existing oidc (keycloakUrls aren't persisted, so if we have issuer & authEndpoint set custom endpoints to true)
+      // Cover use case where user edits existing oidc (oidcUrls aren't persisted, so if we have issuer & authEndpoint set custom endpoints to true)
       if (!old && neu) {
-        this.customEndpoint.value = (!this.keycloakUrls.url && !this.keycloakUrls.authEndpoint) && (!!this.model.issuer && !!this.model.authEndpoint);
+        this.customEndpoint.value = (!this.oidcUrls.url && !this.oidcUrls.authEndpoint) && (!!this.model.issuer && !!this.model.authEndpoint);
       }
     }
   },
 
   methods: {
-    updateIssuerEndpoint() {
-      if (!this.keycloakUrls.url) {
+    updateEndpoints() {
+      if (!this.oidcUrls.url) {
         return;
       }
-      const url = this.keycloakUrls.url.replaceAll(' ', '');
+      const url = this.oidcUrls.url.replaceAll(' ', '');
 
-      this.model.issuer = `${ url }/auth/realms/${ this.keycloakUrls.realm || '' }`;
+      this.model.issuer = `${ url }/realms/${ this.oidcUrls.realm || '' }`;
       this.model.authEndpoint = `${ this.model.issuer || '' }/protocol/openid-connect/auth`;
```

Comment:
following on from https://github.com/rancher/dashboard/pull/11112#discussion_r1633350312, for generic oidc this now isn't mandatory and we shouldn't pre-populate it (which means the backend will do discovery)

Reply:
Update: I just gave this a try and get the following error for /v3/keyCloakOIDCConfigs/keycloakoidc?action=configureTest

```
{"baseType":"error","code":"NotNullable","fieldName":"authEndpoint","message":"","status":422,"type":"error"}
```

@crobby was there an updated image that supports this? does discovery work if both issuer and authEndpoint are supplied (we're sending an empty string for the later, but not sure that would cause the error)?

Reply:
My mistake.  I will push a new image that allows for an empty auth endpoint.